NonSucking.Framework.Serialization/Serializers/PublicPropertySerializer.cs

```diff
@@ -17,132 +21,528 @@ namespace NonSucking.Framework.Serialization
+                .MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                    SyntaxFactory.IdentifierName(typeSymbol.ToDisplayString()),
+                    SyntaxFactory.IdentifierName(genMethod.Name));
+
+            statements.Statements.Add(
+                SyntaxFactory.ExpressionStatement(
+                    SyntaxFactory.InvocationExpression(memberAccess,
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SeparatedList(arguments)))));
+        }
+
+        private static List<ArgumentSyntax> GetArgumentsFromGenMethod(
+            string readerName,
+            List<string> declerationNames,
+            GeneratedMethod genMethod)
+        {
+            List<ArgumentSyntax> arguments = new()
+                        {SyntaxFactory.Argument(SyntaxFactory.IdentifierName(readerName))};
+            foreach (var item in genMethod.Parameters.Skip(1))
             {
-                prop.ScopeVariableNameMappings = scopeVariableNameMappings;
-                var propCode = GenerateStatementsForMember(prop, context, MethodType.Deserialize);
+                string variableName = item.Name;
+                declerationNames.Add(variableName);
+                arguments.Add(SyntaxFactory
+                    .Argument(null,
+                        SyntaxFactory.Token(SyntaxKind.OutKeyword),
+                        SyntaxFactory.DeclarationExpression(
+                            SyntaxFactory.IdentifierName(
+                                SyntaxFactory.Identifier(
+                                    SyntaxFactory.TriviaList(),
+                                    SyntaxKind.VarKeyword,
+                                    "var",
+                                    "var",
+                                    SyntaxFactory.TriviaList())),
+                            SyntaxFactory.SingleVariableDesignation(
+                                SyntaxFactory.Identifier(variableName)))));
+            }
 
-                if (propCode is null)
-                    continue;
-                foreach (var item in propCode.VariableDeclarations)
+            return arguments;
+        }
+
+        private static GeneratedMethod GetOrCreateGenMethod(
+            NoosonGeneratorContext context,
+            string readerName,
+            IEnumerable<GeneratedMethodParameter> leadingMethodParameters,
+            (List<(string typeName, string parameterName)> Parameters, string typeName, IMethodSymbol? Symbol)? hasBaseDeserialize,
+            Dictionary<string, string> scopeVariableNameMappings,
+            List<MemberInfo> filteredProps,
+            List<(bool assign, GeneratedSerializerCode propCode)> propCodes,
+            ITypeSymbol typeSymbol,
+            GeneratedType genType)
+        {
+            GeneratedMethod? genMethod =
+                genType.Methods.FirstOrDefault(x => x.Name == Consts.Deserialize /*&& x.Typestuff*/);
```

Comment:
```suggestion
                genType.Methods.FirstOrDefault(x => x.Name == Consts.Deserialize /* TODO: && x.Typestuff*/);
```
so that we can find it again atleast